cmd/osmosisd/cmd/forceprune.go

```diff
@@ -0,0 +1,145 @@
+package cmd
+
+// DONTCOVER
+
+import (
+  "github.com/spf13/cobra"
+
+  "fmt"
+  "strconv"
+
+  "github.com/syndtr/goleveldb/leveldb"
+  "github.com/syndtr/goleveldb/leveldb/opt"
+  "github.com/syndtr/goleveldb/leveldb/util"
+  tmstore "github.com/tendermint/tendermint/store"
+  tmdb "github.com/tendermint/tm-db"
+
+  "os/exec"
+
+  "github.com/cosmos/cosmos-sdk/client"
+  "github.com/tendermint/tendermint/config"
+)
+
+var fheight = "full_height"
+var mheight = "min_height"
+
+// get cmd to convert any bech32 address to an osmo prefix.
+func forceprune() *cobra.Command {
+  cmd := &cobra.Command{
+    Use:   "forceprune",
+    Short: "Example osmosisd forceprune -f 188000 -m 1000, which would keep blockchain and state data of last 188000 blocks (approximately 2 weeks) and ABCI responses of last 1000 blocks.",
+    Long: "Forceprune options prunes and compacts blockstore.db and state.db. One needs to shut down chain before running forceprune. By default it keeps last 188000 blocks (approximately 2 weeks of data) blockstore and state db (validator and consensus information) and 1000 blocks of abci responses from state.db. Everything beyond these heights in blockstore and state.db is pruned. ABCI Responses are stored in index db and so redundant especially if one is running pruned nodes. As a result we are removing ABCI data from state.db aggressively by default. One can override height for blockstore.db and state.db by using -f option and for abci response by using -m option. Example osmosisd forceprune -f 188000 -m 1000. ",
+    RunE: func(cmd *cobra.Command, args []string) error {
+
+      full_height_flag, err := cmd.Flags().GetString(fheight)
+      if err != nil {
+        return err
+      }
+
+      min_height_flag, err := cmd.Flags().GetString(mheight)
+      if err != nil {
+        return err
+      }
+
+      clientCtx := client.GetClientContextFromCmd(cmd)
+      conf := config.DefaultConfig()
+      db_path := string(clientCtx.HomeDir) + "/" + string(conf.DBPath)
+
+      cmdr := exec.Command("osmosisd", "status")
+      err = cmdr.Run()
+
+      if err == nil {
+        // continue only if throws errror
+        return nil
+      }
```

Comment:
We can delete these lines right, they were just there when it was intended to be a standalone go script?

Reply:
This was to check and silently fail further processing if chain was up.
If chain is up leveldb.OpenFile will fail.